File: c/hello.rb
#!/usr/bin/env ruby

def help_msg()
	puts "'h' go left"
	puts "'l' go right"
	puts "'?' this help message"
	puts "'q' quit"
end

def room_msg(x)
  puts "you are in room #{x}"
end

room = 0
game_loop = true

help_msg()
room_msg(room)

while game_loop do
  case gets.chomp
  when 'h'
    room -= 1
    room_msg(room)
  when 'l'
    room += 1
    room_msg(room)
  when '?'
    help_msg()
  when 'q'
    game_loop = false
  else
    # do nothing
  end
end

puts "thanks for playing"

File: c/hello.c
#include <stdio.h>
#include <stdbool.h>

void room_msg(int x);
void help_msg(void);

int main(void)
{
	int room = 0;
	int game_loop = 1;

	help_msg();
	room_msg(room);
	while (game_loop) {
		switch (getchar()) {
		case 'h':
			room--;
			room_msg(room);
			break;
		case 'l':
			room++;
			room_msg(room);
			break;
		case '?': help_msg(); break;
		case 'q': game_loop = 0; break;
		default: break;
		}
	}

	printf("thanks for playing\n");

	return 0;
}

void room_msg(int x)
{
	printf("you are in room %d\n", x);
}

void help_msg(void)
{
	printf("'h' go left\n");
	printf("'l' go right\n");
	printf("'?' this help message\n");
	printf("'q' quit\n");
}
